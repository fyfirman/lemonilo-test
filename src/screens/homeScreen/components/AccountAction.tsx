import React from 'react';
import { ViewProps } from 'react-native';
import styled from 'styled-components/native';

const Container = styled.TouchableOpacity`
  background-color: #fff;
  border-radius: ${({ theme }) => theme.borderRadius.small}px;
  padding: 4px 8px;
`;

const Label = styled.Text`
  font-size: 9px;
  font-weight: bold;
  color: ${({ theme }) => theme.palette.text.primary};
`;

interface AccountActionProps extends ViewProps {
  label: string;
  onPress?: () => any;
}

const AccountAction = ({ label, onPress, ...rest }: AccountActionProps) => {
  return (
    <Container onPress={onPress} {...rest}>
      <Label>{label}</Label>
    </Container>
  );
};

export default AccountAction;
