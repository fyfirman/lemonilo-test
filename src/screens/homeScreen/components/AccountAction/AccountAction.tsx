import myTheme from '@theme/theme';
import React from 'react';
import { ViewProps } from 'react-native';
import styled from 'styled-components/native';

const Container = styled.Pressable`
  background-color: #fff;
  border-radius: ${({ theme }) => theme.borderRadius.small}px;
  padding: 4px 8px;
`;

const Label = styled.Text`
  font-size: 11px;
  font-weight: bold;
  color: ${({ theme }) => theme.palette.text.primary};
`;

interface AccountActionProps extends ViewProps {
  label: string;
  onPress?: () => any;
}

const AccountAction = ({ label, onPress, ...rest }: AccountActionProps) => {
  return (
    <Container
      onPress={onPress}
      android_ripple={{ color: myTheme.palette.ripple.grey, radius: 30 }}
      {...rest}
    >
      <Label>{label}</Label>
    </Container>
  );
};

export default AccountAction;
