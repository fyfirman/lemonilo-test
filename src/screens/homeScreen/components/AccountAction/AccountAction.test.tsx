import React from 'react';
import { renderWithTheme } from '../../../../__tests__/helpers';
import AccountAction from './AccountAction';
import 'jest-styled-components';

test('should match with snapshot', () => {
  const tree = renderWithTheme(<AccountAction label="My Point" />);

  expect(tree).toMatchSnapshot();
});
