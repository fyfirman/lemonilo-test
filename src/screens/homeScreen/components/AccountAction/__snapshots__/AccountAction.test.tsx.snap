// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 4,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0,0,0,0.8)",
          "fontSize": 11,
          "fontWeight": "bold",
        },
      ]
    }
  >
    My Point
  </Text>
</View>
`;
